src/main/java/com/powsybl/sld/server/GeoDataService.java

```diff
@@ -0,0 +1,139 @@
+        this.restTemplate = restTemplate;
+    }
+
+    private String getGeoDataServerURI() {
+        return this.geoDataServerBaseUri + DELIMITER + GEO_DATA_API_VERSION + DELIMITER;
+    }
+
+    public String getLinesGraphics(UUID networkUuid, String variantId, List<String> linesIds) {
+        UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromHttpUrl(getGeoDataServerURI() + LINES)
+                .queryParam(NETWORK_UUID, networkUuid);
+
+        if (!StringUtils.isBlank(variantId)) {
+            uriComponentsBuilder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
+        }
+
+        if (linesIds != null) {
+            uriComponentsBuilder.queryParam(QUERY_PARAM_LINE_ID, linesIds);
+        }
+
+        var path = uriComponentsBuilder
+                .buildAndExpand()
+                .toUriString();
+
+        return restTemplate.getForObject(path, String.class);
+    }
+
+    public String getSubstationsGraphics(UUID networkUuid, String variantId, List<String> substationsIds) {
+        UriComponentsBuilder uriComponentsBuilder = UriComponentsBuilder.fromHttpUrl(getGeoDataServerURI() + SUBSTATIONS)
+                .queryParam(NETWORK_UUID, networkUuid);
+
+        if (!StringUtils.isBlank(variantId)) {
+            uriComponentsBuilder.queryParam(QUERY_PARAM_VARIANT_ID, variantId);
+        }
+
+        if (substationsIds != null) {
+            uriComponentsBuilder.queryParam(QUERY_PARAM_SUBSTATION_ID, substationsIds);
+        }
+
+        var path = uriComponentsBuilder
+                .buildAndExpand()
+                .toUriString();
+
+        return restTemplate.getForObject(path, String.class);
+    }
+
+    public void assignSubstationGeoData(Network network, UUID networkUuid, String variantId, List<Substation> substations) {
+
+        List<SubstationGeoData> substationsGeoData = GeoDataUtils.fromStringToSubstationGeoData(getSubstationsGraphics(networkUuid, variantId, null), new ObjectMapper());
+        Map<String, Coordinate> substationGeoDataMap = substationsGeoData.stream()
+                .collect(Collectors.toMap(SubstationGeoData::getId, SubstationGeoData::getCoordinate));
+
+        for (Substation substation : substations) {
+            if (network.getSubstation(substation.getId()).getExtension(SubstationPosition.class) == null) {
+                com.powsybl.sld.server.dto.Coordinate coordinate = substationGeoDataMap.get(substation.getId());
+                if (coordinate != null) {
+                    network.getSubstation(substation.getId())
+                            .newExtension(SubstationPositionAdder.class)
+                            .withCoordinate(new com.powsybl.iidm.network.extensions.Coordinate(coordinate.getLat(), coordinate.getLon()))
+                            .add();
+                }
+            }
+
+        }
+    }
+
+    public void assignLineGeoData(Network network, UUID networkUuid, String variantId, List<Line> lines) {
```

Comment:
Lines' positions are not taken in account in the network-area-diagram. You should remove this.
It will lighten the process

Reply:
yes removed